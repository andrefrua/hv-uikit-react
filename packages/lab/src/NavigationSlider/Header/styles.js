import { makeStyles } from "@mui/styles";

const styles = (props) =>
  makeStyles((theme) => ({
    menuContainer: {
      display: "grid",
      gridTemplateColumns: "min-content auto",
      minHeight: props.headerHeight,
      boxShadow: "inset 0 5px 5px -3px rgb(65 65 65 / 12%)",
    },
    button: {
      display: "flex",
      alignItems: "center",
    },
    title: {
      display: "flex",
      alignItems: "center",
      justifyContent: "center",
      minHeight: theme.hv.spacing.sm,
      padding: theme.hvSpacing(0, "xs"),
    },
    justifyContentCenter: { justifyContent: "center" },
  }));

export default styles;
