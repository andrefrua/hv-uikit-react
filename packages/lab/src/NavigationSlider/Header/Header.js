import React from "react";
import { withStyles } from "@mui/styles";
import PropTypes from "prop-types";

import { HvButton, HvTypography } from "@hitachivantara/uikit-react-core";
import { Backwards } from "@hitachivantara/uikit-react-icons";

import useStyles from "./styles";

const Header = ({ id, title, showBackButton, headerHeight, onBackButtonClick }) => {
  const classes = useStyles({ headerHeight })();

  const backButtonClickHandler = () => {
    onBackButtonClick();
  };

  return (
    <div id={id} className={classes.menuContainer}>
      <div className={classes.button}>
        {showBackButton && (
          <HvButton icon onClick={backButtonClickHandler}>
            <Backwards iconSize="XS" />
          </HvButton>
        )}
      </div>

      <div className={`${classes.title} ${classes.justifyContentCenter}`}>
        {title && <HvTypography variant="highlightText">{title}</HvTypography>}
      </div>
    </div>
  );
};

Header.propTypes = {
  /**
   * Id to be applied to the root node of the panel.
   */
  id: PropTypes.string,
  /**
   * The text to be displayed on the header.
   */
  title: PropTypes.string,
  /**
   * Flag to show / hide the back button.
   */
  showBackButton: PropTypes.bool,
  /**
   * The header height.
   */
  headerHeight: PropTypes.number,
  /**
   * Callback function triggered when the back button is clicked.
   */
  onBackButtonClick: PropTypes.func,
};

Header.defaultProps = {
  headerHeight: 80,
};

export default withStyles(useStyles, { name: "Header" })(Header);
