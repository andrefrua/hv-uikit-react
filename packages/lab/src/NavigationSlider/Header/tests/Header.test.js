import React from "react";
import { create } from "react-test-renderer";

import { HvButton, HvTypography, HvProvider } from "@hitachivantara/uikit-react-core";

import Header from "..";

describe("<Header /> with empty values", () => {
  let renderer;

  beforeAll(() => {
    renderer = create(
      <HvProvider>
        <Header />
      </HvProvider>
    );
  });

  it("should match the snapshot", () => {
    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it("should be defined", () => {
    expect(renderer).toBeDefined();
  });
});

describe("<Header /> with title and back button", () => {
  let renderer;
  const mockOnBackButtonClick = jest.fn(() => {});

  beforeAll(() => {
    renderer = create(
      <HvProvider>
        <Header title="DummyTitle" showBackButton onBackButtonClick={mockOnBackButtonClick} />
      </HvProvider>
    );
  });

  it("should display the provided title", async () => {
    const typographyElement = await renderer.root.findByType(HvTypography);
    const titleElement = await typographyElement.findByType("p");

    expect(titleElement.children[0]).toBe("DummyTitle");
  });

  it("should render a back button and clicking on it triggers the callback", async () => {
    const backButton = await renderer.root.findByType(HvButton);
    expect(backButton.instance).toBeDefined();

    backButton.props.onClick();

    expect(mockOnBackButtonClick).toBeCalled();
  });
});
