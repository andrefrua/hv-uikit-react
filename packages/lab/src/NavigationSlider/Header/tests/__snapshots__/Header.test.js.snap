// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> with empty values should match the snapshot 1`] = `
<div
  className="makeStyles-menuContainer-9"
>
  <div
    className="makeStyles-button-10"
  />
  <div
    className="makeStyles-title-11 makeStyles-justifyContentCenter-12"
  />
</div>
`;
